Automattic/jetpack
# VideoPress: re implement useResumableUploader(). Avoid unneeded jwt-upload request

On this PR:

* Re implement the useResumableUploader() hook with TS. It keeps the current implementation until we update all imports.
* Update VideoPress uploader to use this hook
* Update getMediaToken() to support `jwt-upload` one
* Fixes https://github.com/Automattic/jetpack/issues/28131
* Move upload to `resumableFileUploader()`
* More TypeScript Changes


#### Changes proposed in this Pull Request:

* VideoPress: Do not perform jwt-upload request unneeded 

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to block editor
* Create a new VideoPress video block
* Confirm the app does not perform a jwt-upload request
* Confirm it's possible to upload a video file as usual
